tests/TestRunner/app/ApiTests.js

```diff
@@ -465,6 +465,24 @@ describe(module.id, function () {
         expect(TNSReturnsRetained.newNSObjectMethod().retainCount()).toBe(1);
     });
 
+    it("unmanaged", function() {
+      var unmanaged = functionReturnsUnmanaged();
+      expect('takeRetainedValue' in unmanaged).toBe(true);
+      expect('takeUnretainedValue' in unmanaged).toBe(true);
+      expect(functionReturnsUnmanaged().takeRetainedValue().retainCount()).toBe(1);
+
+      var value = functionReturnsUnmanaged().takeUnretainedValue();
+      expect(value.retainCount()).toBe(2);
+      CFRelease(value);
+
+      expect(function() {
+        (function() {
+          unmanaged.takeRetainedValue();
```

Comment:
The first call may be called before the `expect.toThrow` block, so that it doesn't eventually swallow an error if the first call throws. Also, what is the use of the self-executing function inside?